File: 933-number-of-recent-calls/solution.go
package main

type Queue []int

type RecentCounter struct {
	queue Queue
	ridx  int
}

func Constructor() RecentCounter {
	return RecentCounter{
		queue: Queue{},
	}
}

func (this *RecentCounter) Ping(t int) int {
	this.queue = append(this.queue, t)
	for this.queue[this.ridx] < t-3000 {
		this.ridx++
	}
	return len(this.queue) - this.ridx
}

/**
 * Your RecentCounter object will be instantiated and called as such:
 * obj := Constructor();
 * param_1 := obj.Ping(t);
 */

File: 933-number-of-recent-calls/solution.py
from collections import deque

class RecentCounter:

    def __init__(self):
        self.c = deque()

    def ping(self, t: int) -> int:
        self.c.append(t)
        r = [t - 3000, t]
        while self.c and self.c[0] < r[0]:
            self.c.popleft()

        return len(self.c)

# Your RecentCounter object will be instantiated and called as such:
# obj = RecentCounter()
# param_1 = obj.ping(t)